app/src/main/java/src/utilities/GameBoard.java

```diff
@@ -14,42 +41,75 @@ public void init() { }
      * @param width is the game board width
      * @return the game board array
      */
-    public int[][] createGameBoardArray(final int height, final int width) {
-        int[][] gameBoardArray = new int[height][width];
+    public boolean[][] createGameBoardArray(final int height, final int width) {
+        boolean[][] gameBoardArray = new boolean[height][width];
         for (int i = 0; i < height; i++) {
             for (int j = 0; j < width; j++) {
-                gameBoardArray[i][j] = 0;
+                gameBoardArray[i][j] = false;
             }
         }
         return gameBoardArray;
     }
 
     /**
-     * Sets all the values of a row to 0
+     * Sets all the values of a row to false
      * @param gameBoardArray is the array to set the values
      * @param row is the row on the array to set the values
      * @return the array with the values set
      */
-    public int[][] cleanRowOnArray(final int[][] gameBoardArray, final int row) {
+    public boolean[][] cleanRowOnArray(final boolean[][] gameBoardArray, final int row) {
         for (int j = 0; j < gameBoardArray[row].length; j++) {
-            gameBoardArray[row][j] = 0;
+            gameBoardArray[row][j] = false;
         }
         return gameBoardArray;
     }
 
     /**
-     * Verifies if a certain row is full of 1
+     * Verifies if a certain row is full of a certain value
      * @param gameBoardArray is the array to check
      * @param row is the row to be verified
+     * @param arrayValue is the value to check
      * @return a boolean according to the verification made
      */
-    public boolean checkFullRow(final int[][] gameBoardArray, final int row) {
-        int notEmptyValueCounter = 0;
+    public boolean checkFullRow(final boolean[][] gameBoardArray, final int row, final boolean arrayValue) {
+        int valueCounter = 0;
         for (int j = 0; j < gameBoardArray[row].length; j++) {
-            if (gameBoardArray[row][j] == 1) {
-                notEmptyValueCounter += 1;
+            if (gameBoardArray[row][j] == arrayValue) {
+                valueCounter += 1;
+            }
+        }
+        return (valueCounter == gameBoardArray[row].length);
+    }
+
+    /**
+     * Moves the lines to the bottom side when empty lines in between
+     * @param gameBoardArray the array to adjust
+     */
+    public void adjustLines(final boolean[][] gameBoardArray) {
+        int row = gameBoardArray.length - 1;
+        int emptyRowCounter;
+        while (row >= 0) {
+            emptyRowCounter = 0;
+            while (checkFullRow(gameBoardArray, row, false)) {
+                emptyRowCounter += 1;
+                if (row == 0) {
+                    break;
+                } else {
+                    row -= 1;
+                }
+            }
+            if (emptyRowCounter == 0 || row == 0) {
+                row -= 1;
+            } else {
+                boolean[][] array = new boolean[1][gameBoardArray[row].length];
```

Comment:
I think you should change the name "array", since it is not so descriptive

Reply:
Ok, i will change it to new row since it is getting the new row of the array.